[DOCS] Update Result Format Topic (#8674)
great-expectations/great_expectations: docs/docusaurus/docs/reference/expectations/result_format.md

- `partial_unexpected_count`: Sets the number of results to include in `partial_unexpected_count``, if applicable. Set the value to zero to suppress the unexpected counts.

- `include_unexpected_rows`: When running validations, this returns the entire row for each unexpected value in dictionary form. When using `include_unexpected_rows`, you must explicitly specify `result_format` and `result_format` must be more verbose than `BOOLEAN_ONLY`.

    :::note
    `include_unexpected_rows` returns EVERY row for each unexpected value. In large tables, this could result in an unmanageable amount of data.
    :::

## Configure result format

You can specify `result_format` for a single Expectation, or an entire Checkpoint. When configured at the Expectation-level, 
the configuration is not persisted, and you'll receive a `UserWarning`. GX recommends that you use an Expectation-level configuration for exploratory analysis, and add the final configuration at the Checkpoint-level.

### Expectation-level configuration

> To apply `result_format` to an Expectation, you pass it into the Expectation. 
> 
> 1. Run `$ great_expectations suite new` to obtain a Validator object instance.
> 
> 2. Run the following code to pass `result_format` into the Expectation:
> 
>     ```python name="tests/integration/docusaurus/reference/core_concepts/result_format/result_format_complete_example_set"
>     ```

Reviewer: ```suggestion
To apply `result_format` to an Expectation, you pass it into the Expectation configuration on your Validator:

    ```python name="tests/integration/docusaurus/reference/core_concepts/result_format/result_format_complete_example_set"
    ```
```

:::note
The `unexpected_index_list`, as represented by primary key (PK) columns, is 
rendered in Data Docs when `COMPLETE` is selected. 

The `unexpected_index_query`, which for `SQL` and `Spark` is a query that allows you to retrieve the full set of unexpected values from the dataset, is also rendered by default when `COMPLETE` is selected. For `Pandas`, this parameter returns the full set of unexpected indices, which can also be used to retrieve the full set of unexpected values. This is returned
whether the `unexpected_index_column_names` are defined. 

To suppress this output, set the `return_unexpected_index_query` parameter to `False`. 

Regardless of how Result Format is configured, `unexpected_list` is never rendered in Data Docs.
:::

## Expectations result format values and fields

| Fields within `result`                | BOOLEAN_ONLY                       |BASIC           |SUMMARY         |COMPLETE        |
----------------------------------------|------------------------------------|----------------|----------------|-----------------
|    details (dictionary)               | Defined on a per-Expectation basis |

> 

Reviewer: For some reason I can't edit this section, but it should be removed.

Author: @austiezr I have removed the _Expectations result format values and fields_ section.

> ## Column map Expectations result format values and fields

Reviewer: ```suggestion
## Column Map Expectations result format values and fields
```

|    unexpected_index_list              |no              |no              |no              |yes             |
|    unexpected_index_query             |no              |no              |no              |yes             |
|    unexpected_list                    |no              |no              |no              |yes             |

> ## Column aggregate Expectations result format values and fields

Reviewer: ```suggestion
## Column Aggregate Expectations result format values and fields
```

```python name="tests/integration/docusaurus/reference/core_concepts/result_format/result_format_summary_example_agg"
```

Returns the following output:

```python name="tests/integration/docusaurus/reference/core_concepts/result_format/result_format_summary_example_agg_output"
```

### Behavior for `COMPLETE`

A `result` is generated with all available justification for why an Expectation was successful or unsuccessful. The format is intended for debugging pipelines or developing detailed regression tests.

> Great Expectations has standard behavior for support for describing the results of `column_map_expectation` and `ColumnAggregateExpectation` Expectations.

Reviewer 2: This reads a bit awkwardly with the double 'for' in it.  Maybe something like:
```suggestion
Great Expectations implements standard behaviors to support describing the results of `column_map_expectation` and `ColumnAggregateExpectation` Expectations.
```
...not a strong recommendation.  Just seemed the wording here could be better.
